File: 45주차/완주하지못한선수.js
function solution(participant, completion) {
  var answer = "";
  var dict = {};
  for (let name of participant) {
    if (name in dict) dict[name] += 1;
    else dict[name] = 1;
  }
  for (let name of completion) {
    dict[name] -= 1;
  }
  for (let name in dict) {
    if (dict[name] > 0) {
      answer = name;
      break;
    }
  }

  return answer;
}
// 프로그래머스 고득점 kit - 해시 js문제

File: 47주차/완주하지못한선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    map<string,int> p;
    int n = participant.size();
    for (int i = 0; i < n; i++) {
        string name = participant[i];
        if (p.find(name) == p.end()) p[name] = 1;
        else p[name] += 1;
    }
    
    for (int i = 0; i < n-1; i++) {
        string name = completion[i];
        p[name] -= 1;
    }
    
    for (int i = 0; i < n; i++) {
        string name = participant[i];
        if (p[name] == 1) return name;
    }
    
    return answer;
}// 플그머 문제 - 해시